### PR TITLE
Fix T-SQL's IDENTITY_INSERT syntax

### PR DESCRIPTION
**Pull request by fdw**
### Brief summary of the change made

T-SQL's `IDENTITY_INSERT` needs a table name in addition to `ON`/`OFF`. So we add it.

Fixes: #2448

### Are there any other side effects of this change that we should be aware of?
I don't think so 🙂

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_tsql.py [hunk 1]
```diff
@@ -1520,7 +1520,10 @@ class SetStatementSegment(BaseSegment):
             "CONCAT_NULL_YIELDS_NULL",
             "CURSOR_CLOSE_ON_COMMIT",
             "FIPS_FLAGGER",
-            "IDENTITY_INSERT",
+            Sequence(
+                "IDENTITY_INSERT",
+                Ref("TableReferenceSegment")
+            ),
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_tsql.py:1526]: Need to run Black on your code, but suspect this is the change it'll suggest:

```suggestion
            Sequence("IDENTITY_INSERT", Ref("TableReferenceSegment")),
```

**Review by tunetheweb — APPROVED** [review]
LGTM once you run Black

**Comment by fdw** [comment]
> LGTM once you run Black

Was about to, when I saw that you've fixed it already 😉 Thank you!

**Comment by tunetheweb** [comment]
> > LGTM once you run Black
> 
> Was about to, when I saw that you've fixed it already 😉 Thank you!

Took a punt! Looks like I guessed right 😁

BTE the precommit is a life saver for this annoyance: https://github.com/sqlfluff/sqlfluff/blob/main/CONTRIBUTING.md#pre-commit-config

**Pull request merged**